### DISCUSSION TITLE
ESP32-S3 UNO board + 8-bit ILI9341 not working

### DISCUSSION THREAD

**Discussion by twtomato** [Bodmer/TFT_eSPI#3349] [Compatible support libraries]
[ESP32-S3 UNO](https://www.aliexpress.com/item/1005006217206292.html) pin definition is as follows

![cn-11134208-7r98o-lomb64zk8srm21](https://github.com/Bodmer/TFT_eSPI/assets/13249873/7a7ce9bd-2367-43f4-8c07-dac4032f3ada)
![71yn1fzh7dL _SL1001_](https://github.com/Bodmer/TFT_eSPI/assets/13249873/8f273ab2-c4a2-4fad-92bb-44255d6c950a)

```
#define TFT_CS 6
#define TFT_DC 7
#define TFT_RST 5
#define TFT_WR 1
#define TFT_RD 2
#define TFT_D0 21
#define TFT_D1 46
#define TFT_D2 18
#define TFT_D3 17
#define TFT_D4 19
#define TFT_D5 20
#define TFT_D6 3
#define TFT_D7 14
```

The TFT_D1 pin is mapped to GPIO 46, a high GPIO. Could this be causing the initialization failure? Any assistance would be appreciated.

**Comment by twtomato** [comment]
I have temporarily resolved this issue by soldering a jumper wire to connect GPIO46 to GPIO15. However, I believe it can be done programmatically. I still need the assistance of experts to help with this. Thank you all.
![vlcsnap-2024-06-07-03h50m07s722](https://github.com/Bodmer/TFT_eSPI/assets/13249873/896f56e0-33cf-40d5-b595-46bed853275d)


Here is a video of testing NerdMiner.
https://www.youtube.com/watch?v=E_ReicJSFG4

**Reply by tonbor** [reply]
Any other solution then soldering a jumper wire to connect GPIO46 to GPIO15. Did you change TFT_D1 to GPIO 15 in the setting after soldering the jumper wire GPIO46 to GPIO15?

**Reply by twtomato** [reply]
I haven't paid attention to this library for a while, so I don't know if there is any new progress.
Yes, you have to change TFT_D1 to GPIO 15
My new settings:
#define TFT_D0 21
#define TFT_D1 15
#define TFT_D2 18
#define TFT_D3 17
#define TFT_D4 19
#define TFT_D5 20
#define TFT_D6 3
#define TFT_D7 14

**Comment by tonbor** [comment]
I found the problem, this library can only handle a databus in the range from pin 0 to 32 or 32 and above, not a databus with mixed pins. Only the first D0 IS CHECKED in this library for the range. -> #if defined (TFT_PARALLEL_8_BIT)  #if (TFT_D0 >= 32) // If D0 is a high GPIO assume all other data bits are high GPIO.
I switched to the lovyangfx library, there is this not a problem. So not soldering a wire needed.
![ae_1752250618668](https://github.com/user-attachments/assets/4647750d-94bc-4022-91d6-08b41a273a39)
